server/src/main/resources/buildServerResources/editProjectConnectionVault.jsp

```diff
@@ -153,4 +162,20 @@
 <script>
   $j('#OAuthConnectionDialog .popupSaveButtonsBlock .testConnectionButton').remove();
   $j("#testConnectionButton").appendTo($j('#OAuthConnectionDialog .popupSaveButtonsBlock')[0])
+
+  $j(document).ready(function(){
+      let checked = $j('#aws-iam-auth').is(':checked')
+      $j("#approle_auth_endpoint").toggle(!checked);
+      $j("#approle_role_id").toggle(!checked);
+      $j("#approle_secret_id").toggle(!checked);
+  })
+
+  $j('#aws-iam-auth').click(function() {
+      $j("#approle_auth_endpoint").toggle(!this.checked);
```

Comment:
Could be simplified by using class `auth_approle` on all relevant `tr` tags and `$j(".auth_approle").toggle(!this.checked)`